### DISCUSSION TITLE
Better screen resolution / fonts as a "Modernization" issue

### DISCUSSION THREAD

**Discussion by masinter** [Interlisp/medley#215] [Ideas]
I am finally seeing the requirements that lead to people wanting better screen resolution as a "modernization" issue.
Medley was designed to run on old vacuum-tube monitors with circuits that caused a kind of blurriness, and on today's LCD color displays, the result is something difficult to stand.  I'm not sure if changing fontset and increasing the screen size, pixel doubling, or reviving XMAS will do it for us, but this is "must do". In the short term, I think adding -fg -bg colors are a good start.

I feel better about the possibility of addressing system-wide issues now that we have makeinit (still needs a cleanup).

**Comment by blakemcbride** [comment]
More important to me is the maximum size and initial size specifications, two different issues as follows:

1. As Nick noted, there is a maximum window size baked into the system.  This is too small for many of the modern, large monitors.
2. Specifying the initial window size and its maximized size, parameters passed on the command line, do not work (as I detailed previously).

**Comment by masinter** [comment]
Do not discount the value to visibility that comes from proper use of outline fonts.   This is all common now.  I don't know about Medley color, but I do know that all the outline font work happened later.
Fonts are a difficult art.  We would do better to use platform rendering for \displayline. Maybe that was the ultimate destination?

**Reply by nbriggs** [reply]
All the nice fonts come from using a 24-bpp color display (or 8-bpp monochrome) rather than a 1-bpp display.  A platform drawline on a 1-bpp display will give you the same as Interlisp does now (those algorithms haven't changed much), and if you're drawing on a multiple bit-per-pixel display you won't be able to read them back to Lisp (unless we fix up the multi bpp code) which seems fundamental to the Interlisp display handling philosophy... ?

**Comment by masinter** [comment]
The NeXT used 2 bits per pixel (black, white, 2 levels of grey). By \DISPLAYLINE I was referring to the (TEdit) display-a-line-of-text code, not drawing a line.

**Reply by nbriggs** [reply]
I agree, 2-bpp is better than 1-bpp.  Current font (and graphics) rendering takes advantage of the extra depth, and also takes account of the ordering of the RGB sub-pixels on a color display.  We still have the issue of, for example, `(SNAPW)` needing to read back the display buffer.   Here's an example of "Ca" rendered in full black against a white background...
![PastedGraphic-1](https://user-images.githubusercontent.com/8138841/109404598-83126c80-791c-11eb-8e93-781f681045df.png)

**Comment by masinter** [comment]
i think 8 bit-per-pixel is probably simpler  than 24. I don't think there is too much that relies on 1 bit per pixel in the SCREENBITMAP. There was apparently some move to model Interlisp bitmaps as Common Lisp 2-D arrays. Sometimes I forget this is Lisp code.
